app/models/mixins/aggregation_mixin/methods.rb

```diff
@@ -39,11 +39,11 @@ def all_storages
     end
 
     def aggregate_hardware(from, field, targets = nil)
-      from      = from.to_s.singularize
+      from      = from.to_s.singularize.to_sym
       select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
       targets ||= send("all_#{from}_ids")
       targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
```

Comment:
Pretty sure you can get rid of this  line too.

rails will automatically call `targets.to_param` on models that are passed to active record. (which is basically a `id` lookup)